homeassistant/components/hassio/repairs.py

```diff
@@ -0,0 +1,138 @@
+"""Supervisor events monitor."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.issue_registry import (
+    IssueSeverity,
+    async_create_issue,
+    async_delete_issue,
+)
+
+from .const import (
+    ATTR_DATA,
+    ATTR_HEALTHY,
+    ATTR_SUPPORTED,
+    ATTR_UNHEALTHY,
+    ATTR_UNHEALTHY_REASONS,
+    ATTR_UNSUPPORTED,
+    ATTR_UNSUPPORTED_REASONS,
+    ATTR_UPDATE_KEY,
+    ATTR_WS_EVENT,
+    DOMAIN,
+    EVENT_HEALTH_CHANGED,
+    EVENT_SUPERVISOR_EVENT,
+    EVENT_SUPERVISOR_UPDATE,
+    EVENT_SUPPORTED_CHANGED,
+    UPDATE_KEY_SUPERVISOR,
+)
+from .handler import HassIO
+
+ISSUE_ID_UNHEALTHY = "unhealthy_system"
+ISSUE_ID_UNSUPPORTED = "unsupported_system"
+
+INFO_URL_UNHEALTHY = "https://www.home-assistant.io/more-info/unhealthy"
+INFO_URL_UNSUPPORTED = "https://www.home-assistant.io/more-info/unsupported"
+
+
+class SupervisorRepairs:
+    """Create repairs from supervisor events."""
+
+    def __init__(self, hass: HomeAssistant, client: HassIO) -> None:
+        """Initialize supervisor repairs."""
+        self._hass = hass
+        self._client = client
+        self._unsupported_reasons: set[str] = set()
+        self._unhealthy_reasons: set[str] = set()
+
+    @property
+    def unhealthy_reasons(self) -> set[str]:
+        """Get unhealthy reasons. Returns empty set if system is healthy."""
+        return self._unhealthy_reasons
+
+    @unhealthy_reasons.setter
+    def unhealthy_reasons(self, reasons: set[str]) -> None:
+        """Set unhealthy reasons. Create or delete repairs as necessary."""
+        for unhealthy in reasons - self.unhealthy_reasons:
+            async_create_issue(
+                self._hass,
+                DOMAIN,
+                f"{ISSUE_ID_UNHEALTHY}_{unhealthy}",
+                is_fixable=False,
+                learn_more_url=f"{INFO_URL_UNHEALTHY}/{unhealthy}",
+                severity=IssueSeverity.CRITICAL,
+                translation_key="unhealthy",
+                translation_placeholders={"reason": unhealthy},
+            )
+
+        for fixed in self.unhealthy_reasons - reasons:
+            async_delete_issue(self._hass, DOMAIN, f"{ISSUE_ID_UNHEALTHY}_{fixed}")
+
+        self._unhealthy_reasons = reasons
+
+    @property
+    def unsupported_reasons(self) -> set[str]:
+        """Get unsupported reasons. Returns empty set if system is supported."""
+        return self._unsupported_reasons
+
+    @unsupported_reasons.setter
+    def unsupported_reasons(self, reasons: set[str]) -> None:
+        """Set unsupported reasons. Create or delete repairs as necessary."""
+        for unsupported in reasons - self.unsupported_reasons:
+            async_create_issue(
+                self._hass,
+                DOMAIN,
+                f"{ISSUE_ID_UNSUPPORTED}_{unsupported}",
```

Comment:
Does this mean that we’re creating a repair issue per unsupported reason ? I guess that’s done to make sure the learn more URL works. I wonder if we should make a “landing page” that creates a list of links to all the unsupported things passed in.

```python
f”{INFO_URL_UNSUPPORTED}?menu={‘,’.join(unsupported)}”
```

Reply:
I'm fine with that. Also another option is to allow multiple more info links in repairs.